pkg/secret/secret_test.go

```diff
@@ -1 +1,508 @@
 package secret
+
+import (
+	"fmt"
+
+	"testing"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	defaultSecretName      = "test-name"
+	defaultSecretNamespace = "test-namespace"
+	defaultSecretType      = "test-secretType"
+)
+
+func TestSecretPull(t *testing.T) {
+	testCases := []struct {
+		secretName          string
+		secretNamespace     string
+		expectedError       bool
```

Comment:
You can remove this bool argument and use
if expectedErrorText != "" condition in test case.